internal/controllers/provisioningrequest_hwprovision.go

```diff
@@ -77,23 +122,20 @@ func (t *provisioningRequestReconcilerTask) createNodePoolResources(ctx context.
 // waitForHardwareData waits for the NodePool to be provisioned and update BMC details
 // and bootMacAddress in ClusterInstance.
 func (t *provisioningRequestReconcilerTask) waitForHardwareData(ctx context.Context,
-	clusterInstance *siteconfig.ClusterInstance, nodePool *hwv1alpha1.NodePool) (bool, bool, error) {
+	clusterInstance *siteconfig.ClusterInstance, nodePool *hwv1alpha1.NodePool) (bool, *bool, bool, error) {
 
-	provisioned, timedOutOrFailed, err := t.checkNodePoolProvisionStatus(ctx, nodePool)
-	if provisioned && err == nil {
-		t.logger.InfoContext(
-			ctx,
-			fmt.Sprintf(
-				"NodePool %s in the namespace %s is provisioned",
-				nodePool.GetName(),
-				nodePool.GetNamespace(),
-			),
-		)
-		if err = t.updateClusterInstance(ctx, clusterInstance, nodePool); err != nil {
-			err = fmt.Errorf("failed to update the rendered cluster instance: %w", err)
-		}
+	var configured *bool
+	provisioned, timedOutOrFailed, err := t.checkNodePoolProvisionStatus(ctx, clusterInstance, nodePool)
+	if err != nil {
+		return provisioned, nil, timedOutOrFailed, err
 	}
-	return provisioned, timedOutOrFailed, err
+	if provisioned {
+		configured, timedOutOrFailed, err = t.checkNodePoolConfigStatus(ctx, nodePool)
+	} else {
+		// If the node pool is not provisioned, set configured to nil
+		configured = nil
+	}
```

Comment:
nit: not needed since `configured` is a pointer, its value is nil if not set in the checkNodePoolConfigStatus.

Reply:
Yeah, will remove it in the next update.